### PR TITLE
1518 | Add missing events argument to enec and disec methods

### PR DESCRIPTION
**Pull request by pons93**
Resolves [1518](https://focusuy.atlassian.net/browse/BMC2-1518).

## How to test

1. Activate a python virtual environment.
2. Run `pip install .`
3. Run `python examples\i3c_ibi_example.py` using these methods to (de)activate ibis:
    - `toggle_ibi()`
    - `ccc_unicast_disec()`/`ccc_unicast_enec()`
    - `ccc_broadcast_disec()`/`ccc_broadcast_enec()`

Note: to do this, comment and uncomment the correspondent methods in lines 77...79, 99...101 and 105...107 in `i3c_ibi_example.py`.

### CODE REVIEW

**Review by fausto-focus — APPROVED** [review]
LGTM!

File: examples/i3c_ibi_example.py [hunk 1]
```diff
@@ -73,7 +74,9 @@ def handle_ibi(name, message):
 
     device.on_notification(name="ibi", filter_func=is_ibi, handler_func=handle_ibi)
 
-    i3c.toggle_ibi(target_address, False)
+    #i3c.toggle_ibi(target_address, False)
+    #i3c.ccc_unicast_disec(target_address, [DISEC.DISINT])
+    i3c.ccc_broadcast_disec([DISEC.DISINT])
```
Comment (focus-martin) [inline examples/i3c_ibi_example.py:79]: I would add a comment describing here that there are 3 ways to disable IBIs:
- toggle_ibi method
- Direct DISEC CCC
- Broadcast DISEC CCC

File: examples/i3c_ibi_example.py [hunk 2]
```diff
@@ -93,10 +96,16 @@ def handle_ibi(name, message):
     i3c.write(target_address, i3c.TransferMode.I3C_SDR, [0x76], [0x00])
     i3c.write(target_address, i3c.TransferMode.I3C_SDR, [0x4E], [0x02])
 
-    i3c.toggle_ibi(target_address, True)
+    #i3c.toggle_ibi(target_address, True)
```
Comment (focus-martin) [inline examples/i3c_ibi_example.py:99]: The same here regarding disabling IBIs

File: supernovacontroller/sequential/i3c.py [hunk 3]
```diff
@@ -885,14 +885,17 @@ def ccc_rstdaa(self, target_address):
...
+    def ccc_broadcast_enec(self, events: list):
         """
         Performs a broadcast ENEC (Enable Events Command) operation on the I3C bus.
 
         This method sends a broadcast command to enable events on all devices on the I3C bus.
         The operation's success status is checked, and it returns a tuple indicating whether the operation
         was successful along with the relevant data or error message.
 
+        Args:
+        events: A list containing events to be enabled.
```
Comment (focus-martin) [inline supernovacontroller/sequential/i3c.py:897]: Please, mention that the type of element of the list belongs to the enum ENEC: ``ENEC.ENINT``, ``ENEC.ENCR``, ``ENEC.ENHJ``. For instance [ENEC.ENINT, ENEC.ENHJ]

File: supernovacontroller/sequential/i3c.py [hunk 4]
```diff
@@ -913,14 +917,17 @@ def ccc_broadcast_enec(self):
...
+    def ccc_broadcast_disec(self, events: list):
         """
         Performs a broadcast DISEC (Disable Events Command) operation on the I3C bus.
 
         This method sends a broadcast command to disable events on all devices on the I3C bus.
         The operation's success status is checked, and it returns a tuple indicating whether the operation
         was successful along with the relevant data or error message.
 
+        Args:
+        events: A list containing events to be disabled.
```
Comment (focus-martin) [inline supernovacontroller/sequential/i3c.py:929]: Please, mention that the type of element of the list belongs to the enum DISEC: ``DISEC.DISINT``, ``DISEC.DISCR``, ``DISEC.DISHJ``. For instance [DISEC.DISINT, DISEC.DISHJ]

File: supernovacontroller/sequential/i3c.py [hunk 5]
```diff
@@ -951,6 +959,7 @@ def ccc_unicast_enec(self, target_address):
 
         Args:
         target_address: The address of the target device on the I3C bus to which the ENEC command is directed.
+        events: A list containing events to be enabled.
```
Comment (focus-martin) [inline supernovacontroller/sequential/i3c.py:962]: Please, mention that the type of element of the list belongs to the enum ENEC: ENEC.ENINT, ENEC.ENCR, ENEC.ENHJ. For instance [ENEC.ENINT, ENEC.ENHJ]

File: supernovacontroller/sequential/i3c.py [hunk 6]
```diff
@@ -982,6 +992,7 @@ def ccc_unicast_disec(self, target_address):
 
         Args:
         target_address: The address of the target device on the I3C bus to which the DISEC command is directed.
+        events: A list containing events to be disabled.
```
Comment (focus-martin) [inline supernovacontroller/sequential/i3c.py:995]: Please, mention that the type of element of the list belongs to the enum DISEC: DISEC.DISINT, DISEC.DISCR, DISEC.DISHJ. For instance [DISEC.DISINT, DISEC.DISHJ]

**Review by focus-martin — CHANGES_REQUESTED** [review]
JP, I requested some changes to improve the docstring and example documentation.

File: examples/i3c_ibi_example.py (hunk 1 above)
Comment (pons93) [reply examples/i3c_ibi_example.py:79]: Thanks for the suggestions! All have been done!

**Review by Salonso1602 — APPROVED** [review]
LGTM

File: examples/i3c_ibi_example.py [hunk 7]
```diff
@@ -73,7 +74,14 @@ def handle_ibi(name, message):
 
     device.on_notification(name="ibi", filter_func=is_ibi, handler_func=handle_ibi)
 
-    i3c.toggle_ibi(target_address, False)
+    # Supernova SDK offers 3 ways of disabling IBIs:
```
Comment (focus-martin) [inline examples/i3c_ibi_example.py:77]: I would replace Supernova SDK with Supernova Controller SDK

File: examples/i3c_ibi_example.py [hunk 8]
```diff
@@ -93,10 +101,21 @@ def handle_ibi(name, message):
     i3c.write(target_address, i3c.TransferMode.I3C_SDR, [0x76], [0x00])
     i3c.write(target_address, i3c.TransferMode.I3C_SDR, [0x4E], [0x02])
 
-    i3c.toggle_ibi(target_address, True)
+    # Supernova SDK offers 3 ways of enabling IBIs:
```
Comment (focus-martin) [inline examples/i3c_ibi_example.py:104]: The same here

**Pull request merged**